File: beer_quiz/lib/models/quiz_question.dart
class QuizQuestion {
  final String question;
  final List<String> choices;
  final int correctAnswerIndex;
  final String trivia;
  final String difficulty;

  QuizQuestion({
    required this.question,
    required this.choices,
    required this.correctAnswerIndex,
    required this.trivia,
    required this.difficulty,
  });

  factory QuizQuestion.fromJson(Map<String, dynamic> json) {
    return QuizQuestion(
      question: json['question'] as String,
      choices: List<String>.from(json['choices'] as List),
      correctAnswerIndex: json['correctAnswerIndex'] as int,
      trivia: json['trivia'] as String,
      difficulty: json['difficulty'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'question': question,
      'choices': choices,
      'correctAnswerIndex': correctAnswerIndex,
      'trivia': trivia,
      'difficulty': difficulty,
    };
  }
}

File: BeerQuizSwift/BeerQuiz/Models/QuizQuestion.swift
import Foundation

struct QuizQuestion: Codable, Equatable, Identifiable {
    var id: String { question }

    let question: String
    let choices: [String]
    let correctAnswerIndex: Int
    let trivia: String
    let difficulty: Difficulty

    var correctChoice: String {
        choices[correctAnswerIndex]
    }
}
